### PR TITLE
App graph is never None but may be empty

### PR DESCRIPTION
**Pull request by Christian-B**
In the release 6.0.0 if there was no app vertices the algorithms would be passed a None

No app vertices can come from either GFE machine level only or an empty graph.

The no executor branch changed the behaviour to always pass the app graph even if empty.

This made the algorithms that just for loop the app graph a little cleaner.

The issue was the algorithms that have an app and a machine version.

As there is separate work which may remove the machine_graph or at least require there always to be an app graph the choice is to change the if app_grap is None test ti ones that check n_vertices.

If this PR is refused ASB must revert to checking app_graph n_vertices before passing it.

### CODE REVIEW

File: spinn_front_end_common/interface/interface_functions/insert_edges_to_extra_monitor_functionality.py [hunk 1]
```diff
@@ -67,7 +65,7 @@ def __call__(self, machine_graph, placements, machine,
             "Inserting edges between vertices which require FR speed up "
             "functionality.")
 
-        if application_graph is None:
+        if application_graph == 0:
```
Comment (rowleya) [inline spinn_front_end_common/interface/interface_functions/insert_edges_to_extra_monitor_functionality.py:68]: Probably want to check the property here

File: spinn_front_end_common/interface/interface_functions/insert_extra_monitor_vertices_to_graphs.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def __call__(
                 progress, machine, machine_graph, vertex_to_chip_map)
 
         # progress data receiver for data extraction functionality
-        if application_graph is not None:
+        if application_graph.n_verticese > 0:
```
Comment (rowleya) [inline spinn_front_end_common/interface/interface_functions/insert_extra_monitor_vertices_to_graphs.py:68]: Mis-spelled

**Review by rowleya — CHANGES_REQUESTED** [review]

File: spinn_front_end_common/interface/interface_functions/insert_edges_to_extra_monitor_functionality.py (hunk 1 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/interface_functions/insert_edges_to_extra_monitor_functionality.py]: good spot. test got it too Fixed.

File: spinn_front_end_common/interface/interface_functions/insert_extra_monitor_vertices_to_graphs.py (hunk 2 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/interface_functions/insert_extra_monitor_vertices_to_graphs.py]: Yep. Tests got that too. Thanks.

**Comment by Christian-B** [comment]
Will merge with another pr and run jenkins

**Review by rowleya — APPROVED** [review]

**Pull request merged**